circuits/sybil/sybilUtils.circom

```diff
@@ -0,0 +1,104 @@
+    signal output out;
+    out <== 0xbb67ae85;
+}
+
+template GetStateSecretSchemaHash(){
+    signal output schemaHash;
+    schemaHash <== 301485908906857522017021291028488077057;
+}
+
+
+template VerifyAndExtractValStateSecret(holderLevels, gistLevels){
+    signal input claim[8];
+    signal input claimIssuanceMtp[holderLevels];
+    signal input claimIssuanceClaimsRoot;
+    signal input claimIssuanceRevRoot;
+    signal input claimIssuanceRootsRoot;
+    signal input claimIssuanceIdenState; // aka idenGistState
+
+    // signal input claimSchema;
+
+    signal input idenGistMtp[gistLevels];
+    // signal input idenGistState;              // double check this declration 
+    signal input gistRoot;
+
+    signal input userId;
+
+    // inter
+    signal expectedClaimIdx;
+    signal stateSecretSchema;
+
+    signal output secret;
+    // (1) Verify claim is included in claims tree root
+    component claimIssuanceCheck = checkClaimExists(holderLevels);
+    for (var i=0; i<8; i++) { claimIssuanceCheck.claim[i] <== claim[i]; }
+    for (var i=0; i<holderLevels; i++) { claimIssuanceCheck.claimMTP[i] <== claimIssuanceMtp[i]; }
+    claimIssuanceCheck.treeRoot <== claimIssuanceClaimsRoot;
+
+    // verify state includes claims tree
+    component verifyClaimIssuanceIdenState = checkIdenStateMatchesRoots();
+    verifyClaimIssuanceIdenState.claimsTreeRoot <== claimIssuanceClaimsRoot;
+    verifyClaimIssuanceIdenState.revTreeRoot <== claimIssuanceRevRoot;
+    verifyClaimIssuanceIdenState.rootsTreeRoot <== claimIssuanceRootsRoot;
+    verifyClaimIssuanceIdenState.expectedState <== claimIssuanceIdenState;
+
+    component stateSecretSchemaHash = GetStateSecretSchemaHash();
+    stateSecretSchema <== stateSecretSchema.schemaHash;
+
+    // (2) Verify claim schema
+    component claimSchemaCheck = verifyCredentialSchema();
+    for (var i=0; i<8; i++) { claimSchemaCheck.claim[i] <== claim[i]; }
+    claimSchemaCheck.schema <== stateSecretSchema;
+
+    // (3) Verify issuer state is in GIST 
+    component claimIssuanceCheckGISTInc = checkClaimExists(gistLevels);
+    // claimIssuanceIdenState
+    // for (var i=0; i<8; i++) { claimIssuanceCheckGISTInc.claim[i] <== claimIssuanceIdenState[i]; }
+    for (var i=0; i<gistLevels; i++) { claimIssuanceCheckGISTInc.claimMTP[i] <== idenGistMtp[i]; }
+    claimIssuanceCheckGISTInc.treeRoot <== gistRoot;
+
+    component genesisIDhash = Poseidon(1);
+    genesisIDhash.inputs[0] <== userId;
+
+    component gistCheck = SMTVerifier(gistLevels);
+    gistCheck.enabled <== 1;
+    gistCheck.fnc <== 1; // non-inclusion in case if genesis state, otherwise inclusion
+    gistCheck.root <== gistRoot;
+    for (var i=0; i<onChainLevels; i++) { gistCheck.siblings[i] <== gistMtp[i]; }
+    // gistCheck.oldKey <== gistMtpAuxHi;
+    // gistCheck.oldValue <== gistMtpAuxHv;
+    // gistCheck.isOld0 <== gistMtpNoAux;
+    gistCheck.key <== genesisIDhash.out;
+    gistCheck.value <== claimIssuanceIdenState;
```

Comment:
3 comments here:
1) Should be `gistCheck.fnc <== 0` but not `gistCheck.fnc <== 1`. You should check the inclusion
2) This inputs should work here but not be commented out:
gistCheck.oldKey <== 0;
gistCheck.oldValue <== 0;
gistCheck.isOld0 <== 0; 
3) Consider moving the selected part of the code to a separate template and use it in every place where it is repeated. Or, may be, we already have the template.

Reply:
Done